### PR TITLE
feat(unlock-js) - refactor customHookService

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — APPROVED** [review]

**Comment by julien51** [comment]
I still think this is the wrong approach and we should add that logic to web3Service...

**Review by julien51 — CHANGES_REQUESTED** [review]
Please add this to web3Service.

**Comment by searchableguy** [comment]
@julien51 shouldn't it be walletService though given you already have signer by default there and it is the one used for transaction.

**Comment by kalidiagne** [comment]
> @julien51 shouldn't it be walletService though given you already have signer by default there and it is the one used for transaction.

already changed, but since we are going to use add more custom hooks, maybe the things is using the new custom service as before

File: packages/unlock-js/src/web3Service.ts [hunk 1]
```diff
@@ -963,4 +964,49 @@ export default class Web3Service extends UnlockService {
       return ''
     }
   }
+
+  /**
+   * Set signer for `Password hook contract`
+   */
+  async setPasswordHookSigner(
```
Comment (julien51) [inline packages/unlock-js/src/web3Service.ts:971]: This should be in WalletService as it modifies the state.

File: packages/unlock-js/src/web3Service.ts [hunk 2]
```diff
@@ -963,4 +964,49 @@ export default class Web3Service extends UnlockService {
...
+      signer,
+    })
+    const tx = await contract.setSigner(lockAddress, signerAddress)
+    return tx
+  }
+
+  /**
+   * Get signer for `Password hook contract`
+   */
+  async getPasswordHookSigners(
```
Comment (julien51) [inline packages/unlock-js/src/web3Service.ts:995]: This should remain here as it only reads the state

File: packages/unlock-js/src/web3Service.ts (hunk 1 above)
Comment (kalidiagne) [reply packages/unlock-js/src/web3Service.ts]: moved to `WalletService `

**Review by julien51 — APPROVED** [review]

**Pull request merged**